firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/Serial/Nonin/Sensor.h

```diff
@@ -15,35 +15,43 @@
 #include "Device.h"
 #include "Pufferfish/Driver/Initializable.h"
 #include "Pufferfish/HAL/Interfaces/Time.h"
+#include "Pufferfish/Util/Timeouts.h"
 
 namespace Pufferfish::Driver::Serial::Nonin {
 
 struct SensorConnections {
-  bool nonin_connected;
   bool sensor_disconnected;
   bool sensor_alarm;
-  bool spo2_out_of_track;
-  bool hr_out_of_track;
+  bool out_of_track;
 };
 
 /**
  * High-level (stateful) driver for Nonin OEM III SpO2 sensor
  */
 class Sensor : public Initializable {
  public:
-  explicit Sensor(Device &device) : device_(device) {}
+  explicit Sensor(Device &device, HAL::Interfaces::Time &time) : device_(device), time_(time) {}
 
   InitializableState setup() override;
   InitializableState output(float &spo2, float &hr);
-  InitializableState measure(Sample measurements);
 
  private:
-  static bool find(const Flags &measurement, const bool &expected);
+  static const uint32_t waiting_duration = 2;  // ms
```

Comment:
Rename to "measurement_timeout", "sample_timeout", or something similar. Also, this should be 2000 ms, not 2 ms.